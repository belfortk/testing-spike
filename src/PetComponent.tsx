import React from 'react';

interface IProps{
  name: string;
  animal: string;
  breed: string;
}

const PetComponent = ({name, animal, breed} : IProps) => {
  return (
    <div className="pet">
      <h3>{name}</h3>
      <h4>{animal} - {breed}</h4>
    </div>
  );
}

export default PetComponent;