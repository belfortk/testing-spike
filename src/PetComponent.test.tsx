import React from 'react';
import {shallow, render, mount} from "enzyme";
import PetComponent from "./PetComponent";

describe("Pet Component", () =>{
  it("should render pet text with props", () => {
    const wrapper = shallow(<PetComponent name="Nate" animal="Cat" breed="Hairless"/>);
    expect(wrapper).toMatchSnapshot();
    
    expect(wrapper.find("h3")).toHaveLength(1);
    expect(wrapper.find("h4")).toHaveLength(1);
    expect(wrapper.text()).toMatch("NateCat - Hairless");
    wrapper.setProps({name:"Leo"});
    expect(wrapper.text()).toMatch("LeoCat - Hairless");
    wrapper.unmount();
  })
});