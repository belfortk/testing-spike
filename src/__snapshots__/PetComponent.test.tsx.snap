// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pet Component should render pet text with props 1`] = `ShallowWrapper {}`;
